crates/extrinsics/src/lib.rs

```diff
@@ -303,6 +303,44 @@ pub fn parse_code_hash(input: &str) -> Result<<DefaultConfig as Config>::Hash> {
     Ok(arr.into())
 }
 
+/// Calculate the total contract deposit.
+async fn get_contract_total_deposit(
+    storage_base_deposit: Balance,
+    storage_item_deposit: Balance,
+    storage_byte_deposit: Balance,
+    client: &Client,
+) -> Result<Balance> {
+    let contract_pallet_version = fetch_contracts_pallet_version(client).await?;
+    let mut contract_deposit = storage_base_deposit
+        .saturating_add(storage_item_deposit)
+        .saturating_add(storage_byte_deposit);
+
+    // From contracts pallet version 10 deposit calculation has changed.
+    if contract_pallet_version >= 10 {
+        let existential_deposit_address =
+            api::constants().balances().existential_deposit();
```

Comment:
This call introduces a dependency on the Balances pallet. The drawback is that we now have an additional dependency in the metadata, which has caused its size to grow.

Reply:
I see no problem with that. We are not particularly concerned with the binary sized of `cargo-contract`